dataline-scheduler/src/main/java/io/dataline/scheduler/SchedulerPersistence.java

```diff
@@ -57,4 +59,6 @@ long createSyncJob(SourceConnectionImplementation sourceImplementation,
    */
   List<Job> listJobs(JobConfig.ConfigType configType, String configId) throws IOException;
 
+  Optional<Job> getLastSyncJobForConnectionId(UUID connectionId) throws IOException;
```

Comment:
NIT: `getLastSyncJob(UUID connectionId)`. You know it is for a connection id since you have an argument